tests/patch-esmock.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * @fileoverview Run by npm to replace esmock with a legacy release on older versions of Node.js.
+ * @author Francesco Trotta
+ */
+
+import { promises as fs } from "fs";
+
+// esmock < 2.4 supports only Node.js < 19. esmock >= 2.4 supports only Node.js >= 18.6.
+// If we are running Node.js < 19, replace the esmock dependency with the legacy release.
+if (process.versions.modules < 111) {
```

Comment:
out of curiosity: why not using `process.versions.node`? what's the 111? 🤔

Reply:
111 is the ABI version number of Node.js 19, they're all listed at https://nodejs.org/en/download/releases.
`process.versions.node` seems more intuitive, and it should be fine to use it, although it's a string, so we'll need semver to do the comparison.